### ISSUE TITLE
Remove last_plan_solution_ in PlanningComponent

### ISSUE THREAD

**Issue by sjahr** [ros-planning/moveit2#1862]
Let us remove last_plan_solution_ from the PlanningComponent class once #1835 is merged. With the `execute()` method removed, the only actual usage of last_plan_solution_ within the class is removed. It would simplify this class and I think it is better to let the user of PlanningComponent handle the lifetime of results from `plan()` rather than having a member in the planning component who's state might be unknown.
What do you think @henningkayser @AndyZe ?

**Comment by AndyZe** [comment]
I'd agree with that because it seems this is not used anywhere:

```
const planning_interface::MotionPlanResponse& PlanningComponent::getLastMotionPlanResponse()
{
  return last_plan_solution_;
}
```

**Comment by henningkayser** [comment]
I like this idea a lot. Without a meaningful state the only real use case of the PlanningComponent is building of motion plan requests, which should in turn be addressed by #1455. I think that way the API would become much easier to use and to extend.

**Comment by sjahr** [comment]
I'll create a PR. Small annoyance: This will definitely break the API since
```
[[deprecated]]
bool PlanningComponent::execute(bool blocking);
```
wont work anymore

**Comment by sjahr** [comment]
Closed with #1879
